Automattic/wc-calypso-bridge
# Restrict Payment gateways on premium plans

This PR restricts available payment methods for premium users, allowing only WCPay and core Woocommerce gateways.

#### Testing Instructions

- Install an extra payment gateway on a local install. e.g. [paypal](https://wordpress.org/plugins/woocommerce-paypal-payments/)
- In wp-config.php shim APD and set some defines so that plan lookups go through your shim:
```
define( 'IS_ATOMIC', true );
define( 'ATOMIC_CLIENT_ID', 2 ); // WP-on-atomic client id
define( 'ATOMIC_SITE_ID', 149647132 ); // wooptest3.wpcomstaging.com

// Shim APD
class Atomic_Persistent_Data {
	public $site_sticker_woop = true;
	public $WPCOM_PLAN = 'free';
	public function __get( $name ) {
		return null;
	}
}
```
- *Test:* Alter the WPCOM_PLAN variable to different plan levels, `business`, `ecommerce`, `free`, `premium`, `personal`. Only business and ecommerce plans should display paypal.
- *Test:* Disable the woop feature entirely by changing the site_sticker_woop value to `null`.

#### Unittests
```
./docker/phpunit
```

#### Before:
![Screenshot 2021-09-17 at 13-32-53 WooCommerce settings ‹ test — WordPress](https://user-images.githubusercontent.com/811776/133720119-e8a16f04-c385-4a38-8327-5599f30b3fc5.png)

#### After:
![Screenshot 2021-09-17 at 13-33-22 WooCommerce settings ‹ test — WordPress](https://user-images.githubusercontent.com/811776/133720112-2ddb216c-bdcc-4970-bcff-e1632f060e75.png)
